include/samplers/samplers.h

```diff
@@ -425,4 +439,51 @@ void hit_and_run_coord_update(Point &p,
     p.set_coord(rand_coord, p[rand_coord] + bpair.first + kapa * (bpair.second - bpair.first));
 }
 
+
+template <class ConvexBody, class Point, class Parameters, typename NT>
+void billiard_walk(ConvexBody &P, Point &p, NT diameter, std::vector<NT> &Ar, std::vector<NT> &Av, NT &lambda_prev,
+                   Parameters &var, bool first = false) {
+
+    typedef typename Parameters::RNGType RNGType;
+    unsigned int n = P.dimension();
+    RNGType &rng = var.rng;
+    boost::random::uniform_real_distribution<> urdist(0, 1);
+    NT T = urdist(rng) * diameter;
+    Point v = get_direction<RNGType, Point, NT>(n), p0 = p;
+    int it = 0;
+
+    if (first) {
+
+        std::pair<NT, int> pbpair = P.line_positive_intersect(p, v, Ar, Av);
+        if (T <= pbpair.first) {
+            p = (T * v) + p;
+            lambda_prev = T;
+            return;
+        }
+        lambda_prev = 0.995 * pbpair.first;
```

Comment:
`0.995` is this for robustness? maybe define it as  `const` variable

Reply:
done